frontend/src/lib/stores/strats.ts

```diff
@@ -31,61 +35,120 @@ export const initialState = [
     submittedBy: "0xD75004941Dd01B737f04D2C5c94AE16AC32032eF",
     data: [90, 5, 5],
   },
-  // {
-  //   id: "1",
-  //   submittedBy:
-  //     "0x054478333586193D177413591900B73AB73dC2F6552e9cD3006f61EC75B2CC2f",
-  //   data: [
-  //     {
-  //       name: "rocket pool",
-  //       val: 60,
-  //     },
-  //     {
-  //       name: "euler",
-  //       val: 22,
-  //     },
-  //     {
-  //       name: "uniswap",
-  //       val: 2,
-  //     },
-  //     {
-  //       name: "wrapped-bitcoin",
-  //       val: 2,
-  //     },
-  //     {
-  //       name: "maker",
-  //       val: 2,
-  //     },
-  //     {
-  //       name: "filecoin",
-  //       val: 2,
-  //     },
-  //     {
-  //       name: "aave",
-  //       val: 2,
-  //     },
-  //     {
-  //       name: "curve-dao-token",
-  //       val: 2,
-  //     },
-  //     {
-  //       name: "nexo",
-  //       val: 2,
-  //     },
-  //     {
-  //       name: "the-graph",
-  //       val: 2,
-  //     },
-  //     {
-  //       name: "sandclock",
-  //       val: 2,
-  //     },
-  //   ],
-  // },
+  {
+    id: "7",
+    submittedBy: "0xAf4f3BDe74e49dDF63Dee2a5Df05687e67553D3f",
+    data: [68, 14, 18],
+  },
+  {
+    id: "8",
+    submittedBy: "0xAf4f3BDe74e49dDF63Dee2a5Df05687e67553D3f",
+    data: [68, 14, 18],
+    assets: [
+      {
+        name: "rocket pool",
+        val: 70,
+      },
+      {
+        name: "euler",
+        val: 12,
+      },
+      {
+        name: "uniswap",
+        val: 2,
+      },
+      {
+        name: "wrapped-bitcoin",
+        val: 2,
+      },
+      {
+        name: "maker",
+        val: 2,
+      },
+      {
+        name: "filecoin",
+        val: 2,
+      },
+      {
+        name: "aave",
+        val: 2,
+      },
+      {
+        name: "curve-dao-token",
+        val: 2,
+      },
+      {
+        name: "nexo",
+        val: 2,
+      },
+      {
+        name: "the-graph",
+        val: 2,
+      },
+      {
+        name: "sandclock",
+        val: 2,
+      },
+    ],
+  },
 ];
 
-export type Strategy = typeof initialState[0];
+export type Strategy = {
+  id: string;
+  submittedBy: string;
+  data: number[];
+  assets?: { name: string; val: number }[];
+};
+
+function strategiesStore() {
+  const treasuryContract = get(contracts).treasury;
+
+  const { subscribe, update } = writable<Strategy[]>(initialState);
+
+  async function createStrategy(
+    assets: {
+      name: string;
+      val: number;
+    }[]
+  ) {
+    const strategy = reduceStrategy(assets);
+
+    const tuple = strategy.map(BigNumber.from);
+
+    const { from } = await get(treasuryContract).createStrategy(tuple);
```

Comment:
I'm getting an error here, saying `get(treasuryContract)` is undefined. I think it's because the treasuryContract is only created when we connect Metamask, so at the time this function is created,it's still undefined. You could add a subscription to the `contracts` store to update the value of treasuryContract when the store updates, I think that would fix it.

Reply:
I never got this error, but I added the subscription now, can you you check if it's still happening?

Reply:
That's weird, but it's working for me now 🎉